airflow-core/src/airflow/dag_processing/manager.py

```diff
@@ -413,20 +413,35 @@ def _queue_requested_files_for_parsing(self) -> None:
 
     @provide_session
     def _get_priority_files(self, session: Session = NEW_SESSION) -> list[DagFileInfo]:
-        files: list[DagFileInfo] = []
+        files: set[DagFileInfo] = set()
         bundles = {b.name: b for b in self._dag_bundles}
         requests = session.scalars(
             select(DagPriorityParsingRequest).where(DagPriorityParsingRequest.bundle_name.in_(bundles.keys()))
         )
         for request in requests:
             bundle = bundles[request.bundle_name]
-            files.append(
-                DagFileInfo(
-                    rel_path=Path(request.relative_fileloc), bundle_name=bundle.name, bundle_path=bundle.path
+            if request.parse_whole_folder():
+                # If relative_fileloc is null, get all files from DagModel
+                dag_files = session.scalars(
```

Comment:
We probably should refresh the bundle and [relist the dags from disk](https://github.com/apache/airflow/blob/bf0bfe9dc1f3812989acfe6ed7118acf9ba5b586/airflow-core/src/airflow/dag_processing/manager.py#L579) instead of querying the db.

I think we can just refresh here, relist, and add, but it'd be worth giving it a bit more thought to make sure we aren't impacting other stuff by doing so.